src/core/vectortile/qgsvectortilelayer.cpp

```diff
@@ -119,6 +127,12 @@ bool QgsVectorTileLayer::loadDataSource()
   }
 
   setCrs( QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:3857" ) ) );
+
+  const QgsDataProvider::ProviderOptions providerOptions;
```

Comment:
You should set the transform context here to match the layer's transform context